lib/storage/providers/LocalForage.js

```diff
@@ -57,12 +62,37 @@ const provider = {
      * @return {Promise<void>}
      */
     multiMerge(pairs) {
-        const tasks = _.map(pairs, ([key, value]) => this.setItemQueue.push({key, value, shouldMerge: true}));
+        const tasks = _.map(pairs, ([key, value, currentValue]) => {
+            if (mergeQueue[key]) {
+                mergeQueue[key].push(value);
+                return Promise.resolve();
+            }
+
+            mergeQueue[key] = [value];
+
+            try {
+                const modifiedData = applyMerge(key, currentValue, mergeQueue);
+
+                // Clean up the write queue so we
+                // don't apply these changes again
+                delete mergeQueue[key];
+
+                this.setItemQueue.push({key, modifiedData, shouldMerge: true}).then(() => Promise.resolve());
+            } catch (error) {
+                Logger.logAlert(`An error occurred while applying merge for key: ${key}, Error: ${error}`);
+            }
+
+            return Promise.resolve();
```

Comment:
This method is not waiting for the write operation to complete, but resolves immediately - the method should resolve only after the write is over, same as the other storage provider implementations